openfeature/telemetry/telemetry.go

```diff
@@ -6,90 +7,99 @@ import (
-	Name       string
+	// Name is the name of the event.
+	// It is always "feature_flag.evaluation".
+	Name string
+	// Attributes represents the event's attributes.
 	Attributes map[string]any
-	Body       map[string]any
+	// Body is the flag's value and is only present if the flag's variant is empty.
+	Body map[string]any
 }
 
+// The OpenTelemetry compliant event attributes for flag evaluation.
 const (
-	// The OpenTelemetry compliant event attributes for flag evaluation.
-	// Specification: https://opentelemetry.io/docs/specs/semconv/feature-flags/feature-flags-logs/
-
-	TelemetryKey       string = "feature_flag.key"
-	TelemetryErrorCode string = "error.type"
-	TelemetryVariant   string = "feature_flag.variant"
-	TelemetryContextID string = "feature_flag.context.id"
-	TelemetryErrorMsg  string = "feature_flag.evaluation.error.message"
-	TelemetryReason    string = "feature_flag.evaluation.reason"
-	TelemetryProvider  string = "feature_flag.provider_name"
-	TelemetryFlagSetID string = "feature_flag.set.id"
-	TelemetryVersion   string = "feature_flag.version"
-
-	// Well-known flag metadata attributes for telemetry events.
-	// Specification: https://openfeature.dev/specification/appendix-d#flag-metadata
-	TelemetryFlagMetaContextId string = "contextId"
-	TelemetryFlagMetaFlagSetId string = "flagSetId"
-	TelemetryFlagMetaVersion   string = "version"
-
-	// OpenTelemetry event body.
-	// Specification: https://opentelemetry.io/docs/specs/semconv/feature-flags/feature-flags-logs/
-	TelemetryBody string = "value"
-
-	FlagEvaluationEventName string = "feature_flag.evaluation"
+	FlagKey          string = "feature_flag.key"
+	ErrorTypeKey     string = "error.type"
+	ResultValueKey   string = "feature_flag.result.value"
+	ResultVariantKey string = "feature_flag.result.variant"
+	ErrorMessageKey  string = "error.message"
+	ContextIDKey     string = "feature_flag.context.id"
+	ProviderNameKey  string = "feature_flag.provider.name"
+	ResultReasonKey  string = "feature_flag.result.reason"
+	FlagSetIDKey     string = "feature_flag.set.id"
+	VersionKey       string = "feature_flag.version"
 )
 
+// FlagEvaluationKey is the name of the feature flag evaluation event.
+const FlagEvaluationKey string = "feature_flag.evaluation"
+
+const (
+	flagMetaContextIDKey string = "contextId"
+	flagMetaFlagSetIDKey string = "flagSetId"
+	flagMetaVersionKey   string = "version"
+)
+
+// CreateEvaluationEvent creates an [EvaluationEvent].
+// It is intended to be used in the `Finally` stage of a [openfeature.Hook].
 func CreateEvaluationEvent(hookContext openfeature.HookContext, details openfeature.InterfaceEvaluationDetails) EvaluationEvent {
 	attributes := map[string]any{
-		TelemetryKey:      hookContext.FlagKey(),
-		TelemetryProvider: hookContext.ProviderMetadata().Name,
+		FlagKey:         hookContext.FlagKey(),
+		ProviderNameKey: hookContext.ProviderMetadata().Name,
 	}
 
+	attributes[ResultReasonKey] = strings.ToLower(string(openfeature.UnknownReason))
```

Comment:
The same as `error.type` but for reason. It's well-defined int the [spec](https://openfeature.dev/specification/types#resolution-reason) and should be the same for all sdks.